src/deepsparse/engine.py

```diff
@@ -284,7 +284,7 @@ def num_streams(self) -> int:
         :return: The max count of streams the current instance can handle
            concurrently.
         """
-        return self._num_streams
+        return self._eng_net.num_streams()
```

Comment:
could we `self._num_streams = self._eng_net.num_streams()` after the `_eng_net` is constructed so there aren't two possible sources of truth?

Reply:
if not already, can we add documentation on how the engine chooses the default `num_streams`?

Reply:
I'd prefer to just delete the instance variable all together. It's no longer used after construction.